feat:Rewrite to async/await

rikokko opened the pull request:
# [課題No9](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#9)
async awaitを使って同じことをやってください。rejectは考慮しないでいいです。問題7をasync awaitを使って書いてください
## [CodeSandBox](https://codesandbox.io/s/quirky-hooks-vo3n8?) :1回目のコミットリンク
## [CodeSandBox](https://codesandbox.io/s/dreamy-waterfall-cz6v7?file=/lesson9.js) :2回目のコミットリンク
## [CodeSandBox](https://codesandbox.io/s/magical-snow-o4r3y) :3回目のコミットリンク
## [CodeSandBox](https://codesandbox.io/s/serene-wing-zl7mt) :4回目のコミットリンク
## [CodeSandBox](https://codesandbox.io/s/cocky-fast-qolff) :最新のコミットリンク



## 実装上、不安なところ、重点的に見てもらいたいところ
- コミットリンクhttps://github.com/rikokko/to_be_frontend/pull/17/commits/69c75a348e88aa1dc9c7562572f56f1fc37ca765
The image is not reflected in the code sandbox, but no issues were found. (I will publicize it once, but I will continue to investigate it myself)
I put the display and deletion of the loading image in the fetchListData function, but I'm wondering if it's appropriate.
showLoadingImg () and deleteLoadingImg () couldn't be used alone because they couldn't execute a function like fetchListData.
- 2回目のコミットリンクhttps://github.com/rikokko/to_be_frontend/commit/89d057a129ace8e76ff92fe4f9f992fd68f8bb89
- 3回目のコミットリンクhttps://github.com/rikokko/to_be_frontend/pull/17/commits/2fd04ddd2e5b3ad6ab2cc11757f48424070e17b3
- 4回目のコミットリンクhttps://github.com/rikokko/to_be_frontend/pull/17/commits/964ee48b58d1d150481ed2b228d6e22629d566e6
- 5回目のコミットリンクhttps://github.com/rikokko/to_be_frontend/pull/17/commits/fa22dfe5ebb4445e986ee17a5f61f58290736933

taka0209 commented on lesson9/lesson9.js:
```diff
@@ -0,0 +1,44 @@
+  })
+  const item = await promise;
+  createList(item)
+  deleteLoadingImg()
```
[IMO]
I also think that it is not desirable for showLoadingImg() and deleteLoadingImg() to exist in fetchListData().
How about separating the parts that fetch the data?

taka0209 commented on lesson9/lesson9.js:
```diff
@@ -0,0 +1,44 @@
+  ul.appendChild(fragment)
+}
+
+async function fetchListData() {
+  showLoadingImg()
+  const promise = new Promise((resolve) => {
+    setTimeout(() => resolve(listItems), 3000)
+  })
+  const item = await promise;
+  createList(item)
+  deleteLoadingImg()
+}
```
[IMO]
I also think that it is not desirable for showLoadingImg() and deleteLoadingImg() to exist in fetchListData().
How about separating the parts that fetch the data?

rikokko replied:
Thank you for your comment! As you said, I thought it was inappropriate to display / delete read images when acquiring data. Removed showLoadingImg () deleteLoadingImg () in fetchListData ().
https://github.com/rikokko/to_be_frontend/commit/89d057a129ace8e76ff92fe4f9f992fd68f8bb89

taka0209 replied:
I thought that it was not ideal for createList() to be included in fetchListData(),since createList() works to build the DOM rather than to fetch data.
I also thought it would be more readable to put createList() outside of fetchListData() and await fetchListData().

rikokko replied:
Thank you for your comment.
Removed createList () from fetchListData ().

I couldn't solve the problem "I thought it would be easier to read if I waited for fetchListData ()", so I would appreciate your advice. I could only think of waiting 3 seconds for a list item with a timeout.

taka0209 replied:
For example, I was thinking of something like the following.

`function createList(item) {
  const fragment = document.createDocumentFragment();
  for (let i = 0; i <= item.length - 1; i++) {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");
    a.textContent = item[i].text;
    a.href = item[i].to;
    img.src = item[i].img;
    img.alt = item[i].alt;
    fragment.appendChild(li).appendChild(a).appendChild(img);
  }
  ul.appendChild(fragment);
}

async function fetchListData() {
  return new Promise((resolve) => {
    setTimeout(() => resolve(listItems), 3000);
  });
}

async function asyncProcessing() {
  showLoading();
  const value = await fetchListData();
  removeLoading();
  createList(value);
}

asyncProcessing();`

I was also satisfied with what Kayano explained, so I think I can write it that way.

rikokko replied:
Thank you for your comment. I never thought there was such a way. This way of writing was cleaner, so I used it.🙇‍♀️

haru-programming commented:
`The image is not reflected in the code sandbox, but no issues were found. (I will publicize it once, but I will continue to investigate it myself)`

On the code sandbox, why don't you try deleting the corresponding img file once and raising it again?
It happens to me sometimes, too, but this procedure fixes it :)!

haru-programming reviewed (commented):
IMO
.DS_Store file is up in Git,
I think this should be written in the ignore file and deleted. 
I often give away irrelevant files too!^^;

suefrontend commented on lesson9/lesson9.js:
```diff
@@ -0,0 +1,44 @@
+  { to: "message.html", img: "mail-icon.png", alt: "画像2", text: "メッセージ" }
+];
+
+function showLoadingImg() {
```
Not sure if you need to create a function to display/remove loading image. For me, I just appended/removed the image to/from DOM, rather than calling them as a function.

rikokko replied:
Thank you for your comment. I felt that the content was not so complete as to create a function. I removed this function and wrote the code directly.
https://github.com/rikokko/to_be_frontend/pull/17/commits/89d057a129ace8e76ff92fe4f9f992fd68f8bb89

rikokko replied:
Removed the image display / delete function from fetchListData (). Changed to delete the image of createList (). How about that?
https://github.com/rikokko/to_be_frontend/pull/17/commits/89d057a129ace8e76ff92fe4f9f992fd68f8bb89

rikokko commented on lesson9/lesson9.js:
```diff
@@ -0,0 +1,44 @@
+  ul.appendChild(loadingImg);
+}
+
+function deleteLoadingImg() {
```
In this fix, I removed the delete function, but from the next time I wanted to use delete depending on the content of the function. thank you very much.

rikokko commented on lesson9/lesson9.js:
```diff
@@ -0,0 +1,44 @@
+}
+
+async function fetchListData() {
+  showLoadingImg()
```
Removed the image display / delete function from fetchListData (). Changed to delete the image of createList (). How about that?
89d057a

suefrontend replied:
Looks good!

rikokko commented:
> IMO .DS_Store file is up in Git, I think this should be written in the ignore file and deleted. I often give away irrelevant files too!^^;

Thanks for your information. Deleted .DS_Store file!

rikokko commented:
> `The image is not reflected in the code sandbox, but no issues were found. (I will publicize it once, but I will continue to investigate it myself)`
> 
> On the code sandbox, why don't you try deleting the corresponding img file once and raising it again? It happens to me sometimes, too, but this procedure fixes it :)!

If you did what you said, it was successfully reflected! thank you very much.

taka0209 commented on lesson9/lesson9.js:
```diff
@@ -0,0 +1,36 @@
+  { to: "bookmark.html", img: "bookmark-icon.png", alt: "画像1", text: "ブックマーク" },
+  { to: "message.html", img: "mail-icon.png", alt: "画像2", text: "メッセージ" }
+];
+const loadingImg = document.createElement('img');
+loadingImg.src = "loading-circle.gif";
+loadingImg.id = "loading-img";
+ul.appendChild(loadingImg);
```
IMO
I think it's better to make these into functions, as you did at first.

suefrontend replied:
I'm curious, may I know why you think it's better?

taka0209 replied:
I thought that would be better for showing and hiding the loading icon again.
Am I perhaps misunderstanding something? If so, I am sorry.

suefrontend replied:
Please don't be sorry! I was just curious about your thinking process. I just thought appending/removing the image from DOM directly was more simple than creating functions.

haru-programming reviewed (approved):
Thanks for the correction. 
My part is Approve!
Please correct Taka-san and Kayano-san for the rest :)!

suefrontend commented:
Approved from me. I'm sorry if I make you confused by my review telling you to remove the functions. I'll leave it to Taka-san.

taka0209 commented on lesson9/lesson9.js:
```diff
@@ -0,0 +1,51 @@
+  createList(item);
+}
+
+fetchListData()
```
I think that this is not nessesary.

rikokko replied:
Thank you for your comment! Delete unnecessary functions!
